daemon/src/db.rs

```diff
@@ -246,159 +232,205 @@ pub async fn load_cfd_by_order_id(
-            orders.leverage as leverage,
-            orders.trading_pair as trading_pair,
-            orders.position as position,
-            orders.origin as origin,
-            orders.liquidation_price as liquidation_price,
-            orders.creation_timestamp as creation_timestamp,
-            orders.term as term,
-            orders.oracle_event_id,
-            cfds.quantity_usd as quantity_usd,
-            cfd_states.state as state
-        from cfds as cfds
-        inner join orders as orders on cfds.order_id = orders.id
-        inner join cfd_states as cfd_states on cfd_states.cfd_id = cfds.id
-        where cfd_states.state in (
+        with ord as (
             select
-              state
-              from cfd_states
-            where cfd_id = cfds.id
-            order by id desc
-            limit 1
+                id as order_id,
+                uuid,
+                trading_pair,
+                position,
+                initial_price,
+                min_quantity,
+                max_quantity,
+                leverage,
+                liquidation_price,
+                creation_timestamp_seconds as ts_secs,
+                creation_timestamp_nanoseconds as ts_nanos,
+                term_seconds as term_secs,
+                term_nanoseconds as term_nanos,
+                origin,
+                oracle_event_id
+            from orders
+        ),
+
+        cfd as (
+            select
+                ord.order_id,
+                id as cfd_id,
+                quantity_usd
+            from cfds
+                inner join ord on ord.order_id = id
+        ),
+
+        state as (
+            select
+                id as state_id,
+                cfd.order_id,
+                cfd.quantity_usd,
+                state
+            from cfd_states
+                inner join cfd on cfd.cfd_id = cfd_states.cfd_id
+            where id in (
+                select
+                    max(id) as id
+                from cfd_states
+                group by (cfd_id)
+            )
         )
-        and orders.uuid = ?
+
+        select
+            ord.uuid as "uuid: crate::model::cfd::OrderId",
+            ord.trading_pair as "trading_pair: crate::model::TradingPair",
+            ord.position as "position: crate::model::Position",
+            ord.initial_price,
+            ord.min_quantity,
+            ord.max_quantity,
+            ord.leverage as "leverage: crate::model::Leverage",
+            ord.liquidation_price,
+            ord.ts_secs as "ts_secs: i64",
+            ord.ts_nanos as "ts_nanos: i32",
+            ord.term_secs as "term_secs: i64",
+            ord.term_nanos as "term_nanos: i32",
+            ord.origin as "origin: crate::model::cfd::Origin",
+            ord.oracle_event_id,
+            state.quantity_usd,
+            state.state
+
+        from ord
+            inner join state on state.order_id = ord.order_id
+
+        where ord.uuid = $1
```

Comment:
Not part of this PR, but thought I mention it:

I was unable to figure out a nice way to reuse statement code. These huge statements basically do the same thing besides the final `where` clause`. Can you think of a way to re-use the code - do you think it would be a good idea?

Reply:
The only thing I can think of is adding SQL views to the DB.

Reply:
FWIW I've tried this already with no luck. It may be possible, but it's not simple nor obvious.